src/services/paras/ParasService.ts

```diff
@@ -413,7 +413,7 @@ export class ParasService extends AbstractService {
 			return null;
 		}
 
-		return now.sub(offset).div(leasePeriod);
+		return new BN(now.sub(offset).div(leasePeriod).toNumber());
```

Comment:
although this practically will likely never overflow, I think our API makes some guarantees that the sanitizer will always spit our numbers as strings. Where exactly are the tests failing?

Reply:
Oh I see the BN is the outer most thing. I would ideally avoid the toNumber here, but practically probably ok

Reply:
actually disregard what I am saying, block number is u32 so no overflow risk

Reply:
So this is where it's a little weird. The return value here for `now.sub(offset).div(leasePeriod)` is no longer an instance of BN when returned. Thus giving back a weird and incorrect wonky `leasePeriodIndex`, ex: We expect `39` back in our tests yet got `27`. 

The `.constructor.name` would return 'BN', but ` data instanceof BN || typeof data === 'number'` would return `false` when running through `sanitizeNumbers`. 

Currently looking for the root cause of it. It's between `@polkadot/common` 8.0.5 -> 8.1.2

Reply:
can you do `.toBn()` on the return values?

Reply:
> can you do .toBn() on the return values?

From what I tried, it doesn't exist on the value.

Reply:
> The return value here for now.sub(offset).div(leasePeriod) is no longer an instance of BN when returned.

I would not expect a division of two `BN`s to return a `BN` so at least that part is not surprising?

Reply:
Looking over [bn.js source](https://github.com/indutny/bn.js/blob/4187ca213e91b41acf72be046072f2dc1f06d0de/lib/bn.js#L2428) it _should_ be a BN, unless they have some wonky stuff happening there where if the output fits in 53 bits it returns a regular number instead?

Could we maybe require a newer version of node.js and switch to the native BigInt instead?

Reply:
So we currently already support BigInt natively by requiring a minimum of Node v14. 

> switch to the native BigInt instead?

A lot of polkadot-js's infrastructure is returning BN values, but yea I agree we could switch to BigInt, but it would take some refactoring which should probably exist in its own PR. 

That being said, I isolated this logic in a sanity check, and when I run it I am getting the expoected behavior as oppose to the above:


```javascript
const main = async () => {
    const api = new ApiPromise({
        provider: new WsProvider('wss://kusama-rpc.polkadot.io')
    });

    await api.isReady
    const leasePeriod = await api.consts.slots.leasePeriod;

    console.log(leasePeriod instanceof BN); // -> True
    console.log(leasePeriod.constructor.name); // -> u32
    console.log(new BN(20000).sub(new BN(0)).div(leasePeriod) instanceof BN) -> true
}
```